src/melee.cpp

```diff
@@ -1376,10 +1377,10 @@ item &player::best_shield()
 bool player::block_hit( Creature *source, body_part &bp_hit, damage_instance &dam )
 {
 
-    // Shouldn't block if player is asleep; this only seems to be used by player.
+    // Shouldn't block if player is asleep or winded ; this only seems to be used by player.
     // TODO: It should probably be moved to the section that regenerates blocks
     // and to effects that disallow blocking
-    if( blocks_left < 1 || in_sleep_state() ) {
+    if( blocks_left < 1 || in_sleep_state() || has_effect( efftype_id( "winded" ) ) ) {
```

Comment:
Maybe we should add check for `has_effect( effect_narcosis )` here too?

Reply:
During testing I have seen a fair amount of NPC dodging Zs while being operated.

Reply:
Totally